kyo-core/jvm/src/main/scala/kyo/Path.scala

```diff
@@ -339,22 +354,55 @@ object Path:
 
     type Part = String | Path
 
-    def apply(path: List[Part]): Path =
-        def loop(path: List[Part], acc: List[String]): List[String] =
-            path match
-                case _: Nil.type =>
-                    acc.reverse
-                case h :: t =>
-                    h match
-                        case h: String =>
-                            loop(t, h.split('/').filter(_.nonEmpty).toList.reverse ::: acc)
-                        case h: Path =>
-                            loop(h.path ::: t, acc)
-        new Path(loop(path, Nil))
+    def apply(parts: List[Part]): Path =
+        val flattened = parts.flatMap {
+            case p if isNull(p) => Nil
+            case s: String      => List(s)
+            case p: Path        => p.path
+        }
+        val javaPath       = if flattened.isEmpty then Paths.get("") else Paths.get(flattened.head, flattened.tail*)
```

Comment:
This is now side effecting... not sure if we can avoid that however.

Reply:
hmm is Java's `Path` side effecting?

Reply:
To my surprise, I don't think it is!